lib/rubocop/cop/style/method_call_with_args_parentheses.rb

```diff
@@ -259,13 +259,16 @@ def legitimate_call_with_parentheses?(node)
             allowed_chained_call_with_parentheses?(node)
         end
 
+        # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
The whole `omit_parentheses` style is based around the many edge-cases where we actually _do_ need the parentheses. I have tried to cater to the `Metrics/CyclomaticComplexity` until now, but the extra helpers I introduce to hide the conditions blow up the line length cop. I think the code will be clearer if we don't measure the cyclomatic complexity for this whole cop and eventually inline the helpers where they are used.

Reply:
Introduction of `range_type?` meaning both `irange_type?` and `erange_type?` is proposed.
https://github.com/rubocop-hq/rubocop/pull/6126#discussion_r255288410

We will be able to remove the disabling cop comment when replaced by this proposal . So I think this is fine for this.

Reply:
https://github.com/rubocop-hq/rubocop/pull/6773 will remove the disabling cop comment.

Reply:
Great!